### PR TITLE
Merge pyjanitor dev branch into robertmitchellv's case-when branch

### PR DESCRIPTION
**Pull request by ericmjl**
@samukweku & @robertmitchellv this is the best I can think of at the moment :joy_cat:.

@robertmitchellv, could you start with resolving the conflict? The only thing that needs to be done should be moving your contribution line into `CONTRIBUTORS.md` instead of leaving it in `CONTRIBUTORS.rst`, I think. After that, there should be no more conflicts.

Finally, once the conflicts are resolved, could you do a Squash Merge rather than a Commit Merge? 

Basically, all I was thinking of here was to ensure that the conflicts are resolved up-front. 

If we're having trouble with this one, that's not a problem - we can always re-start the original PR (https://github.com/pyjanitor-devs/pyjanitor/pull/775#) from scratch.

### CODE REVIEW

**Comment by robertmitchellv** [comment]
@ericmjl absolutely--I'll work on this right now, as I have some time. Thanks for your patience @samukweku 🙏🏻

**Comment by samukweku** [comment]
@robertmitchellv apologies, pinging to see how far you've gone with merging this into your existing code and updating.

**Comment by robertmitchellv** [comment]
@samukweku sorry about that--I thought I had pushed the changes to merge already 😱

I only changed the conflict--I didn't see where to move the contribution line I added from `AUTHORS.rst`. Also, there were some failures, but I'm not sure to what degree impacts this PR.

**Comment by samukweku** [comment]
@robertmitchellv sorry, I've been caught up in some other PR. any luck on this ?
